l1-contracts/contracts/bridge/L1SharedBridge.sol

```diff
@@ -373,8 +385,24 @@ contract L1SharedBridge is IL1SharedBridge, ReentrancyGuard, Ownable2StepUpgrade
     }
 
     /// @dev Generate a calldata for calling the deposit finalization on the L2 bridge contract
-    function _getDepositL2Calldata(bytes32 _assetInfo, bytes memory _assetData) internal view returns (bytes memory) {
-        return abi.encodeCall(IL2Bridge.finalizeDeposit, (_assetInfo, _assetData));
+    function _getDepositL2Calldata(
+        address _l1Sender,
+        bytes32 _assetInfo,
+        bytes memory _assetData
+    ) internal view returns (bytes memory) {
+        if (IL1NativeTokenVault(address(nativeTokenVault)).tokenAddress(_assetInfo) == address(0)) {
```

Comment:
No need to unwrap and wrap the NTV, it is already the correct type

Reply:
done